### PR TITLE
Enhancement/86ept8y1y/pricing page encryption

### PR DESCRIPTION
**Pull request by JezChristian**
Added:
Implementation of Encryption
Data and Conditional Renderings are ready for the backend changes

### CODE REVIEW

File: apps/web-customer/src/components/blocks/HomePageBlocks/Pricing/PricingBlock.tsx [hunk 1]
```diff
@@ -3,37 +3,42 @@ import PricingCard from './PricingComponent/PricingCard'
 import { useBusinessQueryContext } from 'core-library/contexts'
 import { SelectedProductType } from 'core-library/types/global'
 import { useRouter } from 'next/router'
+import { Encryption } from 'core-library/utils/Encryption'
+import { config } from 'core-library/config'
+import { useSessionStorage } from 'core-library/hooks'
```
Comment (JMSevilla) [inline apps/web-customer/src/components/blocks/HomePageBlocks/Pricing/PricingBlock.tsx:8]: Can we prevent importing code repeatedly instead you can follow this code format

`export const useAccessToken = () =>
  useSessionStorage<string | undefined>("accessToken", undefined);
`

File: apps/web-customer/src/components/blocks/HomePageBlocks/Pricing/PricingBlock.tsx [hunk 2]
```diff
@@ -3,37 +3,42 @@ import PricingCard from './PricingComponent/PricingCard'
 import { useBusinessQueryContext } from 'core-library/contexts'
 import { SelectedProductType } from 'core-library/types/global'
 import { useRouter } from 'next/router'
+import { Encryption } from 'core-library/utils/Encryption'
+import { config } from 'core-library/config'
+import { useSessionStorage } from 'core-library/hooks'
 
 interface Props { }
 
+
```
Comment (JMSevilla) [inline apps/web-customer/src/components/blocks/HomePageBlocks/Pricing/PricingBlock.tsx:12]: extra spacing

File: apps/web-customer/src/components/blocks/HomePageBlocks/Pricing/PricingBlock.tsx [hunk 3]
```diff
@@ -3,37 +3,42 @@ import PricingCard from './PricingComponent/PricingCard'
...
-    const router = useRouter();
-
-
+    const [, setEncryptedProduct] = useSessionStorage<string | null>('SessionItem', null);
     const { businessQueryGetAllProducts } = useBusinessQueryContext();
     const { data: ProductData } = businessQueryGetAllProducts(['PricingList'])
+    const router = useRouter();
 
     const handleSelectProduct = (product: SelectedProductType) => {
+        const key = config.value.SECRET_KEY
```
Comment (JMSevilla) [inline apps/web-customer/src/components/blocks/HomePageBlocks/Pricing/PricingBlock.tsx:22]: Is this config new ? why I can't find anything from the latest pull

File: apps/web-customer/src/components/blocks/HomePageBlocks/Pricing/PricingBlock.tsx [hunk 4]
```diff
@@ -44,32 +49,38 @@ export const PricingBlock: React.FC<Props> = (props) => {
...
+                    <div className='w-full px-10 flex flex-col gap-5 mt-8'>
+                        <div className='flex gap-5 w-full justify-center flex-wrap '>
+                            {filteredItems && filteredItems.length > 0 ? filteredItems.slice(0, 2).map((item, index) => (
                                 <div key={index}>
                                     <PricingCard key={index} cardData={item} handleSelectProduct={handleSelectProduct} />
                                 </div>
-                            ))}
+                            )) :
+                                <div className={`bg-gradient-to-tr ${nurseType === 0 ? 'from-green-400 to-green-600 text-curveGray' : 'from-blue to-cyan-700 text-white'} rounded-md shadow-md px-5 py-8 text-lg w-full text-center  font-semibold`}>
+                                    <p>Price Data is Currently Not Available</p>
```
Comment (JMSevilla) [inline apps/web-customer/src/components/blocks/HomePageBlocks/Pricing/PricingBlock.tsx:73]: We shouldn't let the user know that the pricing data isn't available. Change this to `Programs unavailable please reload the page`

**Review by JMSevilla — CHANGES_REQUESTED** [review]

File: apps/web-customer/src/components/blocks/HomePageBlocks/Pricing/PricingBlock.tsx (hunk 3 above)
Comment (JezChristian) [reply apps/web-customer/src/components/blocks/HomePageBlocks/Pricing/PricingBlock.tsx:22]: It is the same config used by BJ in the web-customer, it was in my first Pricing API implementation Ticket

**Review by DropitupQwerty — APPROVED** [review]

**Review by JMSevilla — APPROVED** [review]

**Review by acostales08 — APPROVED** [review]

**Pull request merged**